etcdmain/config.go

```diff
@@ -177,6 +178,7 @@ func NewConfig() *config {
 		// Should never happen.
 		plog.Panicf("unexpected error setting up clusterStateFlag: %v", err)
 	}
+	fs.BoolVar(&cfg.strictReconfigCheck, "strict-reconfig-check", false, "Reject member change if a number of running node is less than quorum for avoid deadlock in a case of wrong peer URLs")
```

Comment:
OK, how about this: Reject membership change if the change can cause potential deadlock